app/src/main/java/com/chess/clock/fragments/TimeControlFragment.java

```diff
@@ -285,6 +281,43 @@ public void onTabReselected(TabLayout.Tab tab) {
         updateUi();
     }
 
+    /**
+     * Method to set auto name only if name was not set by user and only in simple mode.
+     */
+    private void autoNaming() {
+        if (!autoNamingEnabled || advancedMode || editMode) return;
+
+        int minutes = getIntOrZero(minutesEt);
+        int seconds = getIntOrZero(secondsEt);
+        int incrementMinutes = getIntOrZero(incrementMinutesEt);
+        int incrementSeconds = getIntOrZero(incrementSecondsEt);
+
+        if (minutes == 0 && seconds == 0) return;
+
+        StringBuilder builder = new StringBuilder();
+        if (minutes > 0) {
+            builder.append(getString(R.string.x_min, minutes));
+        }
+        if (seconds > 0) {
+            builder.append(" ");
+            builder.append(getString(R.string.x_sec, seconds));
+        }
+        if (incrementMinutes > 0 || incrementSeconds > 0) {
+            builder.append(" | ");
+        }
+        if (incrementMinutes > 0) {
+            builder.append(getString(R.string.x_min, incrementMinutes));
+        }
+        if (incrementSeconds > 0) {
+            builder.append(" ");
+            builder.append(getString(R.string.x_sec, incrementSeconds));
+        }
+
+        String newName = builder.toString();
+        latestAutoName = newName;
+        nameEt.setText(newName);
```

Comment:
The string building part would be great candidate to add test for. Let me know if you want me to help with the setup. 
The minor nitpick for example would be that " 30 sec" has leading space so trimming string would not hurt.